text/processing.py

```diff
@@ -305,3 +311,91 @@ def run(self):
                     f.write(f"{line}\n")
             else:
                 raise NotImplementedError
+
+
+class SplitTextFileJob(Job):
+    def __init__(
+        self,
+        text_file: tk.Path,
+        num_lines_per_split: int,
+        num_text_file_lines: Optional[int] = None,
+        zip_output: bool = True,
+    ):
+        """
+        Job splits a text file into several smaller files.
+
+        https://stackoverflow.com/a/45761990/2062195
+
+        :param text_file: Input text file to be processed.
+        :param num_lines_per_split: Number of lines per split.
+        :param num_text_file_lines: Number of lines in the input text file.
+        :param zip_output: compress the output files.
+        """
+        self.in_text_file = text_file
+        self.num_lines_per_split = num_lines_per_split
+        self.zip_output = zip_output
+
+        if num_text_file_lines is not None:
+            self.num_text_file_lines = num_text_file_lines
+
+            self.num_output_files = self.num_text_file_lines // num_lines_per_split + int(
+                bool(self.num_text_file_lines % num_lines_per_split)
+            )
+        else:
```

Comment:
I now understood that this way is not possible, because the outputs can not be dynamic. I would leave this  else branch as "not implemented yet".